solr/solrj-zookeeper/src/java/org/apache/solr/client/solrj/impl/SolrClientNodeStateProvider.java

```diff
@@ -290,15 +290,15 @@ public SimpleSolrResponse invoke(String solrNode, String path, SolrParams params
       String url = zkClientClusterStateProvider.getZkStateReader().getBaseUrlForNodeName(solrNode);
 
       GenericSolrRequest request = new GenericSolrRequest(SolrRequest.METHOD.POST, path, params);
-      try (var client =
-          new HttpSolrClient.Builder()
-              .withHttpClient(solrClient.getHttpClient())
-              .withBaseSolrUrl(url)
-              .withResponseParser(new BinaryResponseParser())
-              .build()) {
-        NamedList<Object> rsp = client.request(request);
+      request.setResponseParser(new BinaryResponseParser());
+
+      try {
+        NamedList<Object> rsp =
+            cloudSolrClient.getHttpClient().requestWithBaseUrl(url, request::process).getResponse();
```

Comment:
instead of a method reference, why not pass request itself (the method is overloaded)